### ISSUE TITLE
Intorducing Editorconfig

### ISSUE THREAD

**Issue by avichalp** [getsenic/senic-hub#419]
In `senic-hub` repo and specially in `Backend` part we are pretty strict about following the pep8 conventions. To complement the checks that run along with the tests it would be great to have some static analysis in our editors so that it corrects us immediately when we miss the convention. 

[Editorconfig](http://editorconfig.org/) also helps in maintaining consistent style and convention across many editors and IDEs. By using the `.editorconfig` file and a plugin for your editor you can take advantage of this tool.

What do you guys think about using it?

**Comment by TheMeaningfulEngineer** [comment]
Hey, 
think this is a good idea.
Don't see editorconfig helping us much to follow pep8 though (do correct me here, I just did a small check and haven't seen options that it would complain about unused imports or number of spaces after a function).

If that is the case, I would argue it makes more sense to look for a single style file that would cover all the cases. A blog addressing the whole linting in python:

https://blog.sideci.com/about-style-guide-of-python-and-linter-tool-pep8-pyflakes-flake8-haking-pyling-7fdbe163079d 

I use [this](https://github.com/vim-syntastic/syntastic) with vim to try and help me out with the pep8.
Didn't really pay much attention on what was actually set up, as soon as it started pointing to the errors CI might complain about.

@avichalp If you're up for defining a policy for our python backend code, and provide examples and tools on how to follow that policy. I would gladly help you iterate on that.

**Comment by avichalp** [comment]
@TheMeaningfulEngineer 
You are right about editorconfig not helping much with pep8. It helps more in maintaining a consistent general styles across different editors. For example: space v/s tabs, no-blank-lines, no. of chars in one line etc.

For pep8 checks I am not currently aware of any tools that integrates across multiple editors.
For independent editors we can user `flake8` as you pointed out in your comment.

I will also look further if you can agree on a style guide to follow.

**Comment by TheMeaningfulEngineer** [comment]
> I will also look further if you can agree on a style guide to follow.

Are you up for just defining what you think is best and we'll iterate on it?

**Comment by avichalp** [comment]
I would not like to define anything for now because the current state of codebase with respect to style is very inconsistent. However I can suggest that we look at different style guides and follow one that most of us agrees upon. We can start looking at [google python style guide](https://google.github.io/styleguide/pyguide.html).
Would like to discuss of what others think about it or any other style guide that they think can help with consistent style across the repo.
